lib/cucumber/filters/retry.rb

```diff
@@ -0,0 +1,19 @@
+require 'cucumber/core/filter'
+require 'cucumber/running_test_case'
+require 'cucumber/events/bus'
+require 'cucumber/events/after_test_case'
+
+module Cucumber
+  module Filters
+    class Retry < Core::Filter.new(:configuration)
+
+      def test_case(test_case)
+        super 
+
+        configuration.on_event(:after_test_case) do |event|
+          test_case.describe_to(receiver) if event.result.failed?
```

Comment:
Off hand I'm not sure I understand, but I'll take a closer look when I get home. Thanks!

> On Oct 22, 2015, at 2:06 PM, Matt Wynne <notifications@github.com> wrote:
> 
> In lib/cucumber/filters/retry.rb:
> 
> > @@ -0,0 +1,19 @@
> > +require 'cucumber/core/filter'
> > +require 'cucumber/running_test_case'
> > +require 'cucumber/events/bus'
> > +require 'cucumber/events/after_test_case'
> > +
> > +module Cucumber
> > +  module Filters
> > +    class Retry < Core::Filter.new(:configuration)
> > +
> > +      def test_case(test_case)
> > +        super 
> > +
> > +        configuration.on_event(:after_test_case) do |event|
> > +          test_case.describe_to(receiver) if event.result.failed?
> This event listener, once registered, will keep firing even after this test case has been and gone. I think you need to register a listener earlier, and use it collect yourself a Hash of test case results in an instance variable, which you can then check here where you want to retry.
> 
> Does that make any sense?
> 
> —
> Reply to this email directly or view it on GitHub.
>